File: ios/App/App/UniversisPlugin.m
//
//  UniversisPlugin.m
//  App
//
//  Created by Michalis Karipidis on 14/5/24.
//

#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

CAP_PLUGIN(UniversisPlugin, "UniversisScraper",
    CAP_PLUGIN_METHOD(authenticate, CAPPluginReturnPromise);
)

File: ios/App/App/UniversisPlugin.swift
import Capacitor

@objc(UniversisPlugin)
public class UniversisPlugin: CAPPlugin {
    @objc func authenticate(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        print("Chris")
        let credentials = UniversisScraperLogic.scrape(username: call.getString("username") ?? "", password: call.getString("password") ?? "")
        
        call.resolve(["value": credentials])
    }
}
